processor/attributesprocessor/factory.go

```diff
@@ -115,10 +240,12 @@ func createMetricsProcessor(
 		return nil, err
 	}
 
-	skipExpr, err := filtermetric.NewSkipExpr(
-		filtermetric.CreateMatchPropertiesFromDefault(oCfg.Include),
-		filtermetric.CreateMatchPropertiesFromDefault(oCfg.Exclude),
-	)
```

Comment:
@mattsains, thanks for all your effort in fixing it. Overall it looks good to me, but I'm concerned that we scatter the filtering logic that is supposed to be consolidated in `filtermetric` module. 

The original bug ignoring resource attributes is in https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/ddaa6841ed8fc908d296179ec67ebb4356b864d6/internal/filter/filtermetric/filtermetric.go#L51-L60. Filter processor has its own workaround for that https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/processor/filterprocessor/metrics.go#L65-L67. What do you think if we move the resource filtering logic from `filterprocessor` to the `filtermetric` making it available for both processors?

Filtering by attribute is an arguable feature, but if you need it, I think it should be added to both filter and attribute processors - in `filtermetric` module. That can be done in a separate PR to make the changes atomic and easier to review / spot problems.

Please let me know what you think.

Reply:
I can change it for both processors.

I am ambivalent about data point matching, happy to take direction on this. If we decide not to implement it, I will update the README and unit tests to remove references to this non existent feature

Reply:
Thank you.

> I am ambivalent about data point matching, happy to take direction on this. If we decide not to implement it, I will update the README and unit tests to remove references to this non existent feature

Let's start with resource attributes matching first, reusing the functionality available in `filterprocessor` and updating the docs

Reply:
I was looking at the config passed to NewSkipExpr: https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/internal/filter/filtermetric/config.go#L40

how does the ResourceAttributes field interact with MatchType, specifically the Expr match type? It seems like here: https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/internal/filter/filtermetric/expr_matcher.go#L42 the expression matcher only matches on the metric and not the resource, should I augment this to also match resources on expressions? Not sure entirely what that would mean.

Reply:
I think we can start with the existing behavior of the filter processor and support `resource_attributes` for strict and regex modes only

Reply:
I'm working on this. Will update PR as soon as I can

Reply:
I've been thinking, I think one of the main reasons to use an attribute processor is to be able to modify data point attributes and therefore I think it's natural to want to filter them too. From the readme:

> The attributes processor modifies attributes of a metric data point, log record or span event.
> Please refer to config.go for the config spec.
>
> This processor also supports the ability to filter and match input data to determine
> if they should be included or excluded for specified actions.
>
> [...]
>
> The list of actions can be composed to create rich scenarios, such as
> back filling attribute, copying values to a new key, redacting sensitive information.


What are your thoughts on this?